apps/openmw/mwphysics/actor.cpp

```diff
@@ -150,6 +150,16 @@ void Actor::setCanWaterWalk(bool waterWalk)
     {
         mCanWaterWalk = waterWalk;
         updateCollisionMask();
+	if (waterWalk) {
+	    ESM::Position pos = mPtr.getRefData().getPosition();
+	    if (pos.pos[2] < -100) {
+		/* I didn't look for the exact limit but watercolision will
+		 * push the player out of water only if there is not too much
+		 * water, for -100 and beyond position must be adjusted ! */
+		pos.pos[2] = -50;
+		mPtr.getRefData().setPosition(pos);
```

Comment:
Actually even in the vanilla game you could get stuck if using walk on
water in places where there is no room to walk on water !
For the "wrong place", well it seems ok to me since it always happens when
casting "walk on water" in vanilla, but I am opened to suggestions...


2015-09-15 14:13 GMT+02:00 scrawl <notifications@github.com>:

> In apps/openmw/mwphysics/actor.cpp
> <https://github.com/OpenMW/openmw/pull/761#discussion_r39502537>:
>
> > @@ -150,6 +150,16 @@ void Actor::setCanWaterWalk(bool waterWalk)
> >      {
> >          mCanWaterWalk = waterWalk;
> >          updateCollisionMask();
> > +	if (waterWalk) {
> > +	    ESM::Position pos = mPtr.getRefData().getPosition();
> > +	    if (pos.pos[2] < -100) {
> > +		/* I didn't look for the exact limit but watercolision will
> > +		 * push the player out of water only if there is not too much
> > +		 * water, for -100 and beyond position must be adjusted ! */
> > +		pos.pos[2] = -50;
> > +		mPtr.getRefData().setPosition(pos);
>
> Before moving the actor, we should check if there's an obstacle in the
> way, or you could get stuck.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenMW/openmw/pull/761/files#r39502537>.
>